src/asami/entities.cljc

```diff
@@ -174,20 +174,32 @@
                                (if-let [overflow (:overflow (ex-data e))]
                                  (reduced last-result)
                                  (throw e))))
-                           (if (and (seqable? obj)
+                           (if (and (seqable? obj)          ; would it be ok performance-wise to turn this into if-let with destructuring to extract e a v to use below instead of (nth ...)?
                                     (= 4 (count obj))
                                     (= :db/add (first obj)))
                              (or
-                              (when (= (nth obj 2) :db/id)
+                               ;; Ex.: `[:db/add [:id X] :id X]` that creates a new entity
+                               (when-let [ref (and (writer/lookup-ref? (nth obj 1))
```

Comment:
I suggest wrapping this in a `(let [rf (nth obj 1)] ...`

ie.
```clojure
(let [rf (nth obj 1)]
  (when-let [ref (and (writer/lookup-ref? rf)
                      (= (first rf) (nth obj 2))
                      rf)]
```

I'd have called it `ref`, but I didn't want to shadow it :)